### ISSUE TITLE
Concept of Multi GPU Caffe training

### ISSUE THREAD

**Issue by sathish249** [BVLC/caffe#4143]
Hi ALL,
          I have been working on caffe for last five months, did various training with BVLC Caffe.
But i couldn't get the concept of how it runs on Multi-GPU mode.
I have a 4 and 8 NVIDIA GPU in my server compiled with cuDNN v4.0.4. During the training i give ./build/tools/caffe train -gpu 0,1,2,3.    I learned that if its a 4 GPU run... My input Batchsize gets multiplied by 4. i.e 256 Batchsize from the solver file effectively at runtime will become 256X4 = 1024 batchsize.  

I want to understand... will each of the GPU executing 256 batch size is independent of other GPU. or each 256 batchsize is split across 4 GPUs and runs (Data/Model Parallelism) in the BVLC caffe.

Why the batchsize is getting multiplied with number of GPU present? 4 GPU as expected takes minimum 3X more time... What should be the learning rate for 4 GPU approximately to acheive  57% accuracy.

**Comment by shelhamer** [comment]
Please ask usage questions on the mailing list. The effective batch size of Caffe parallelism is batch size * number of GPUs so to parallelize the same number of inputs you need to divide the batch size.

From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
